x-pack/plugins/security_solution/public/common/links/links.tsx

```diff
@@ -0,0 +1,198 @@
+  ...(link.icon != null ? { euiIconType: link.icon } : {}),
+  ...(link.image != null ? { icon: link.image } : {}),
+  id: link.id,
+  ...(link.globalSearchKeywords != null ? { keywords: link.globalSearchKeywords } : {}),
+  ...(link.globalNavEnabled != null
+    ? { navLinkStatus: link.globalNavEnabled ? AppNavLinkStatus.visible : AppNavLinkStatus.hidden }
+    : {}),
+  ...(link.globalNavOrder != null ? { order: link.globalNavOrder } : {}),
+  path: link.url,
+  ...(link.globalSearchEnabled != null ? { searchable: link.globalSearchEnabled } : {}),
+  title: link.label,
+});
+
+const hasFeaturesCapability = (
+  features: Feature[] | undefined,
+  capabilities: Capabilities
+): boolean => {
+  if (!features) {
+    return true;
+  }
+  return features.some((featureKey) => get(capabilities, featureKey, false));
+};
+
+const reduceDeepLinks = (links: LinkItem[], linkProps?: LinkProps): AppDeepLink[] =>
+  links.reduce((deepLinks: AppDeepLink[], link: LinkItem) => {
+    if (
+      linkProps != null &&
+      // exclude link when license is basic and link is premium
+      ((linkProps.isBasic && link.isPremium) ||
+        // exclude link when enableExperimental[hideWhenExperimentalKey] is enabled and link has hideWhenExperimentalKey
+        (link.hideWhenExperimentalKey != null &&
+          linkProps.enableExperimental[link.hideWhenExperimentalKey]) ||
+        // exclude link when enableExperimental[experimentalKey] is disabled and link has experimentalKey
+        (link.experimentalKey != null && !linkProps.enableExperimental[link.experimentalKey]) ||
+        // exclude link when link is not part of enabled feature capabilities
+        (linkProps.capabilities != null &&
+          !hasFeaturesCapability(link.features, linkProps.capabilities)))
+    ) {
+      return deepLinks;
+    }
+    return [...deepLinks, createDeepLink(link, linkProps)];
+  }, []);
+
+export const getInitialDeepLinks = (): AppDeepLink[] => {
+  return appLinks.map((link) => createDeepLink(link));
+};
+
+export const getDeepLinks = (
+  enableExperimental: ExperimentalFeatures,
+  licenseType?: LicenseType,
+  capabilities?: Capabilities
+): AppDeepLink[] => {
+  const isBasic = licenseType === 'basic';
+  return reduceDeepLinks(appLinks, { enableExperimental, isBasic, capabilities });
+};
+
+const flattenLinkItems = (
+  ids: SecurityPageName[],
+  linkItems: LinkItem[]
+): Array<Omit<LinkItem, 'items'>> =>
+  linkItems.reduce((linkItemFound: Array<Omit<LinkItem, 'items'>>, linkItem) => {
+    let topLevelItems = [...linkItemFound];
+    if (ids.includes(linkItem.id)) {
+      // omit items from result
+      const { items, ...rest } = linkItem;
+      topLevelItems = [...topLevelItems, rest];
+    }
+    if (linkItem.items) {
+      topLevelItems = [...topLevelItems, ...flattenLinkItems(ids, linkItem.items)];
+    }
+    return topLevelItems;
+  }, []);
+
+const urlKeys: Array<{ key: UrlStateType; pages: SecurityPageName[] }> = [
```

Comment:
It seems to me that this `key` could be defined inside `appLinks` so we can keep one centralized configuration.